tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/BasicConsumeIntegration.cs

```diff
@@ -44,9 +46,13 @@ public class BasicConsumeIntegration
         /// <param name="arguments">The original arguments setting</param>
         /// <param name="consumer">The original consumer setting</param>
         /// <returns>Calltarget state value</returns>
-        internal static CallTargetState OnMethodBegin<TTarget, TConsumer>(TTarget instance, string queue, bool autoAck, string consumerTag, bool noLocal, bool exclusive, IDictionary<string, object> arguments, TConsumer consumer)
+        internal static CallTargetState OnMethodBegin<TTarget, TConsumer>(TTarget instance, string? queue, bool autoAck, string? consumerTag, bool noLocal, bool exclusive, IDictionary<string, object>? arguments, TConsumer consumer)
         {
-            QueueHelper.SetQueue(consumer, queue);
+            if (consumer is not null && queue is not null)
```

Comment:
nit, I would have done the check in the QueueHelper but this is the only usage so feel free to ignore.

Reply:
It's kinda interesting, as it's a bit of a difference in philosophy. Either
- Change `SetQueue` to accept nullable arguments, and don't do the checks here
- Keep `SetQueue` not accepting nullable arguments, and move the checks here instead.

Given that `QueueHelper` was _already_ marked with `#nullable enable` _and_ it's `internal`, this approach seems the most correct to me. One of the benefit of NRTs is that you _don't_ need to add extra null checks in all your types "just in case". Which means you should push the null checks out to the "edges" of your program, basically anywhere you interface with customer's or third-party libraries where you can't guarantee the nullability...

Which is all a long-winded way of me seeing: I think I'm right here 😛

Reply:
> Which means you should push the null checks out to the "edges" of your program

Ok, seeing it like that, I agree.

> Which is all a long-winded way of me seeing: I think I'm right here 😛

🤣